### PR TITLE
Add subtitles size and offset for StreamItemState

### PR DESCRIPTION
**Pull request by tymmesyde**

### CODE REVIEW

**Review by kKaskak — APPROVED** [review]

File: src/types/streams/streams_item.rs [hunk 1]
```diff
@@ -33,6 +33,10 @@ pub struct StreamItemState {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub subtitle_delay: Option<i64>,
     #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (elpiel) [inline src/types/streams/streams_item.rs]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

File: src/types/streams/streams_item.rs [hunk 2]
```diff
@@ -33,6 +33,10 @@ pub struct StreamItemState {
...
+    pub subtitle_size: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (elpiel) [inline src/types/streams/streams_item.rs]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

**Review by elpiel — CHANGES_REQUESTED** [review]
These are the only changes that should guard against failing deserialization of already existing `StreaamItemState`s in the `StreamsBucket`

File: src/types/streams/streams_item.rs [hunk 3]
```diff
@@ -32,6 +32,10 @@ pub struct StreamItemState {
     /// In milliseconds
     #[serde(skip_serializing_if = "Option::is_none")]
     pub subtitle_delay: Option<i64>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub subtitle_size: Option<u64>,
```
Comment (elpiel) [inline src/types/streams/streams_item.rs:37]:
```suggestion
    /// Subtitles size, platform dependent units
    pub subtitle_size: Option<u64>,
```

File: src/types/streams/streams_item.rs [hunk 4]
```diff
@@ -32,6 +32,10 @@ pub struct StreamItemState {
...
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub subtitle_offset: Option<u64>,
```
Comment (elpiel) [inline src/types/streams/streams_item.rs:40]:
```suggestion
    /// Vertical offset of the subtitles, platform dependent units
    pub subtitle_offset: Option<u64>,
```

**Review by elpiel — APPROVED** [review]

**Pull request merged**